### PR TITLE
Fix for IPVLAN after init.sh partial rewrite in Go

### PR DESCRIPTION
**Pull request by vadorovsky**
My previous PR #13915 introduced a regression in IPVLAN mode. This PR contains the fix and enables IPVLAN in CI.

### CODE REVIEW

**Comment by mrostecki** [comment]
test-me-please

**Review by pchaigno — APPROVED** [review]

File: test/k8sT/DatapathConfiguration.go [hunk 1]
```diff
@@ -319,6 +319,48 @@ var _ = Describe("K8sDatapathConfig", func() {
 		})
 	})
 
+	Context("IPVLAN", func() {
+		BeforeEach(func() {
+			SkipIfIntegration(helpers.CIIntegrationEKS)
+			SkipIfIntegration(helpers.CIIntegrationFlannel)
+			SkipIfIntegration(helpers.CIIntegrationGKE)
+		})
```
Comment (fristonio) [inline test/k8sT/DatapathConfiguration.go]: If I understand the intent correctly then we can achieve this with:

```
SkipContextIf(func() bool {
    return helpers.GetCurrentIntegration() != ""
}, "IPVLAN", func() {})
```

File: test/k8sT/DatapathConfiguration.go [hunk 2]
```diff
@@ -319,6 +319,48 @@ var _ = Describe("K8sDatapathConfig", func() {
...
+
+		It("Check connectivity with IPVLAN", func() {
+			options := map[string]string{
+				"tunnel":              "disabled",
+				"datapathMode":        "ipvlan",
+				"ipvlan.masterDevice": "eth0",
```
Comment (fristonio) [inline test/k8sT/DatapathConfiguration.go:329]: I believe we don't have an `eth0` device in our test VMs. Will this still work?

File: test/k8sT/DatapathConfiguration.go [hunk 3]
```diff
@@ -319,6 +319,48 @@ var _ = Describe("K8sDatapathConfig", func() {
...
+				options["masquerade"] = "false"
+			}
+			deploymentManager.DeployCilium(options, DeployCiliumOptionsAndDNS)
+
+			Expect(testPodConnectivityAcrossNodes(kubectl)).Should(BeTrue(), "Connectivity test between nodes failed")
+			if helpers.RunsOnNetNextOr419Kernel() {
+				By("Test BPF masquerade")
+				Expect(testPodHTTPToOutside(kubectl, "http://google.com", false, false)).
+					Should(BeTrue(), "Connectivity test to http://google.com failed")
+			}
```
Comment (fristonio) [inline test/k8sT/DatapathConfiguration.go]: Sorry, I am not really sure if I understand how this is testing BPF masquerading behavior?

**Review by fristonio — CHANGES_REQUESTED** [review]
Thanks for adding the tests :tada:

File: test/k8sT/DatapathConfiguration.go (hunk 3 above)
Comment (brb) [reply test/k8sT/DatapathConfiguration.go]: The BPF masq is currently not supported in the ipvlan mode.

File: test/k8sT/DatapathConfiguration.go (hunk 2 above)
Comment (brb) [reply test/k8sT/DatapathConfiguration.go:329]: You could use the same logic for detecting devices as we have in `overwriteHelmOptions()` function.

File: test/k8sT/DatapathConfiguration.go (hunk 3 above)
Comment (vadorovsky) [reply test/k8sT/DatapathConfiguration.go]: OK, then I will just remove that test.

**Comment by mrostecki** [comment]
test-me-please

File: test/k8sT/DatapathConfiguration.go [hunk 4]
```diff
@@ -319,6 +319,35 @@ var _ = Describe("K8sDatapathConfig", func() {
...
+		It("Check connecitivity with IPVLAN in pure L3 mode", func() {
+			options := map[string]string{
+				"tunnel":               "disabled",
+				"autoDirectNodeRoutes": "true",
+				"datapathMode":         "ipvlan",
+				"ipvlan.masterDevice":  "eth0",
+				"installIptablesRules": "false",
+				"l7Proxy.enabled":      "false",
+			}
+			deploymentManager.DeployCilium(options, DeployCiliumOptionsAndDNS)
```
Comment (christarazi) [inline test/k8sT/DatapathConfiguration.go:345]: Nit: you can iniline `options` into the `DeployCilium` call

**Review by christarazi — APPROVED** [review]
LGTM. Minor nit below.

**Review by nebril — APPROVED** [review]

**Comment by mrostecki** [comment]
Marked as a draft. There is a chance that we will deprecate ipvlan.

**Comment by brb** [comment]
@mrostecki Can you create a separate PR for the ipvlan fix so that the ipvlan mode can be fixed in master?

**Pull request closed without merging**